### PR TITLE
feat: adjust auth store to standard

### PR DESCRIPTION
**Pull request by walt-it**
# ⚡ {0} - [{1}](https://app.clickup.com/t/{2}) ⚡

## 💻 What type of change is this?

- [{3}] 💎 Feature
- [{4}] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Styling
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build / CI

## ⭐ Description

- Makes all our stores comply with our standard, persisted stores shouldn't be written different as they should behave the same.
- Wasn't able to test it since we don't have a login implemented. But chat GPT says this:

```
Yes, the token will still be persisted to localStorage even though setToken is defined outside the store definition. Here’s why:

Zustand’s setState function: The setToken function simply calls useAuthStore.setState, which is a built-in function provided by Zustand to update the store's state. This method directly interacts with the store.

Persist Middleware: When you use the persist middleware, Zustand wraps the store with additional logic to automatically synchronize any changes made to the store state with localStorage. In this case, since persist is configured to store the state under the key "authStore", any changes to the token (whether done via setToken or any other update function) will be saved to localStorage.

So, even though setToken is defined outside the store, as long as it calls useAuthStore.setState, it updates the state in a way that triggers the persist logic to store the updated token in localStorage.
```

## 📷 Screenshots

n/a

## 💬 Comments

n/a

### Requirements

- [ ] This change includes database migrations
- [ ] This PR requires changes in the `.env` file
- [ ] This PR requires to rebuild a table or run a cron manually
- [x] This PR can be merged (it is not a draft, work in progress, or blocked on another PR)

### To review

- [ ] I have tested this change locally in multiple screen sizes
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] If my task include an endpoint, I add the endpoint to Hopscotch/Postman Project
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/stores/useAuthStore.ts [hunk 1]
```diff
@@ -1,34 +1,27 @@
...
 
 export type AuthStoreState = {
   token: string | null;
-  setToken(token: string | null): void;
 };
 
 export const useAuthStore = create<AuthStoreState>()(
   persist(
-    (set) => ({
+    (_) => ({
```
Comment (walt-it) [inline src/stores/useAuthStore.ts:17]: Not sure why `persist` makes this argument required, but I think this is the official way to bypass it if we are not using it.

File: src/stores/useExampleStore.ts [hunk 2]
```diff
@@ -1,12 +1,8 @@
 /**
- * This file adheres to our standard for Zustand stores without using `persist`.
+ * This file adheres to our standard for Zustand stores.
  * The state and actions are externalized, promoting modularity and separation of concerns,
  * as outlined in Zustand's best practices: https://zustand.docs.pmnd.rs/guides/practice-with-no-store-actions.
  *
- * We use a namespace in the barrel file (e.g., export * as authStore from "./useAuthStore")
```
Comment (walt-it) [inline src/stores/useExampleStore.ts:6]: Now that I started using it, I'm not sure about the barrel. That means everytime we use the store it looks like `authStore.useAuthStore()` which looks a little strange. Typescript will warn you if you try to name two exports the same, so we can rely on that, but I'm open for discussion:

Screenshot 2024-10-03 at 09 40 33

Comment (walt-it) [reply src/stores/useExampleStore.ts:6]: We could name the store hook just `use` under the namespace so usage would be something like:

```
const token = authStore.use(state => state.token)
...
authStore.setToken(newToken)
```

Or even pre-export setters **and getters** like this:

```
export const setToken = (newToken) => useAuthStore.setState({ token: newToken });
export const getToken = () => useAuthStore((state) => state.token)
```

Then usage would be:

```
const token = authStore.getToken()
authStore.setToken(newToken)
```

**Review by ThunderNaka — APPROVED** [review]
Seems fine!

**Comment by walt-it** [comment]
Gonna close and re open with signed commits so I can merge.

**Pull request closed without merging**